src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -299,18 +299,6 @@ ShardIntervalsOnWorkerGroup(WorkerNode *workerNode, Oid relationId)
 			 placementIndex++)
 		{
 			GroupShardPlacement *placement = &placementArray[placementIndex];
-			uint64 shardId = placement->shardId;
```

Comment:
I think this is fine, but we could consider keeping the locks by default, but allowing stale reads if forced:
```SQL
SELECT citus_total_relation_size(regclass, force:=false);
```

I don't have much visibility on the goal of this PR, so @marcocitus might guide us on this.

(And, if `TryLockShardDistributionMetadata ` fails when force==false, we can give a proper error message suggesting to use `force=true`)

Reply:
I don't see the point of these locks given that we don't even have them for regular SELECTs.

Worst case we might in very rare circumstances give a slightly wrong size or fail to get the size.

Reply:
I did not observe any wrong sizes in my tests. The test coverage also appears to not cover the codepath where we intentionally return smaller sizes.

On a failure, the first retry gave me the results most of the time